### DISCUSSION TITLE
How can I disambiguate the foreign key relation with supabasejs?

### DISCUSSION THREAD

**Discussion by yekta** [supabase/supabase#794] [Questions] [answered]
I'm wondering how can I disambiguate the foreign key relation with supabasejs
```
.select(
     `id, title, note, tags:id(id, entry_id, tag_id)`
)
```
I'm querying `entries`. An entry's `id` is referenced in an intermediary table alongside a tag's `id` for a many to many relationship. I'm trying to get all entry `tags` belonging to a particular `entry` while querying entries. But there is another intermediary table which also references the entry's `id` and it causes a problem so I need to disambiguate which one I'm meaning to choose.

<img width="1095" alt="Screen Shot 2021-03-03 at 05 13 20" src="https://user-images.githubusercontent.com/40706601/109742049-635e9b00-7bdf-11eb-8663-2f63210c9855.png">

## EDIT:

After tens of tries, found the solution :D
```
.select(
     `id, title, note, tags:entry_tags.entry_id(id, entry_id, tag_id)`
)
```
`entry_tags` is the intermediary table. `entry_id` is the entry's `id` which is referenced in that intermediary table.

**Comment by steve-chavez** [comment, marked as answer]
Hey @yekta,

Glad you could solve your issue.

I want to point out that the dot syntax `.` in `entry_tags.entry_id` is deprecated.
It'd be better if you'd switch to `!`: `entry_tags!entry_id`.

For more details about ambiguous requests, see: https://postgrest.org/en/v7.0.0/api.html#embedding-disambiguation

**Reply by yekta** [reply]
Thanks for the info, I'll switch to that.
